nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/AWSSensitivePropertyProvider.java

```diff
@@ -0,0 +1,377 @@
+        final String region = awsBootstrapProperties.getProperty(REGION_KEY_PROPS_NAME);
+
+        if (StringUtils.isNotBlank(accessKeyId) && StringUtils.isNotBlank(secretKeyId) && StringUtils.isNotBlank(region)) {
+            logger.debug("Credentials/Configuration provided in bootstrap-aws.conf");
+            try {
+                final AwsBasicCredentials credentials = AwsBasicCredentials.create(accessKeyId, secretKeyId);
+                client = KmsClient.builder()
+                        .region(Region.of(region))
+                        .credentialsProvider(StaticCredentialsProvider.create(credentials))
+                        .build();
+            } catch (final KmsException | NullPointerException | IllegalArgumentException e) {
+                final String msg = "Valid configuration/credentials are required to initialize KMS client";
+                logger.error(msg);
+                throw new SensitivePropertyProtectionException(msg, e);
+            }
+        } else {
+            // attempts to initialize client with credentials provider chain
+            logger.debug("Credentials/Configuration not provided in bootstrap-aws.conf, attempting to use default configuration");
+            try {
+                final DefaultCredentialsProvider credentialsProvider = DefaultCredentialsProvider.builder()
+                        .build();
+                // the following is needed to check the default credential builder, if it fails, throws SdkClientException
+                credentialsProvider.resolveCredentials();
+                client = KmsClient.builder()
+                        .credentialsProvider(credentialsProvider)
+                        .build();
+            } catch (final SdkClientException e) {
+                // this exception occurs if default credentials are not provided
+                final String msg = "Valid configuration/credentials are required to initialize KMS client";
+                logger.error(msg);
+                throw new SensitivePropertyProtectionException(msg, e);
+            }
+        }
+    }
+
+    /**
+     * Checks if the client used to communicate with AWS KMS service is open.
+     * @return true if the client has been initialized and open, false otherwise.
+     */
+    private final boolean isClientOpen() {
+        return client != null;
+    }
+
+    /**
+     * Validates the key ARN, credentials and configuration provided by the user.
+     * Note: This function performs checks on the key and indirectly also validates the credentials and
+     * configurations provided during the initialization of the client.
+     */
+    private final void validate() throws KmsException, SensitivePropertyProtectionException {
+        if (!isClientOpen()) {
+            final String msg = "The AWS KMS Client failed to open, cannot validate key";
+            logger.error(msg);
+            throw new SensitivePropertyProtectionException(msg);
```

Comment:
As recommend elsewhere, recommend removing the log message.
```suggestion
            throw new SensitivePropertyProtectionException("KMS Client not initialized");
```

Reply:
Making the change